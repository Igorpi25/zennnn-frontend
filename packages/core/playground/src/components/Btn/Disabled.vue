<template>
  <div>
    <Btn disabled class="mr-4 mb-6">
      <Icon left>
        {{ ziVisible }}
      </Icon>
      Text icon
    </Btn>
    <Btn disabled class="mr-4 mb-6"> Text </Btn>
    <Btn disabled icon class="mb-6">
      <Icon>
        {{ ziVisible }}
      </Icon>
    </Btn>
  </div>
  <div class="-mb-6">
    <Btn outlined disabled small class="mr-4 mb-6">
      <Icon left>
        {{ ziVisible }}
      </Icon>
      Text icon
    </Btn>
    <Btn outlined disabled small class="mr-4 mb-6"> Text </Btn>
    <Btn outlined disabled icon small class="mb-6">
      <Icon>
        {{ ziVisible }}
      </Icon>
    </Btn>
  </div>
</template>

<script setup>
import { ziVisible } from '@zennnn/icons'
</script>
