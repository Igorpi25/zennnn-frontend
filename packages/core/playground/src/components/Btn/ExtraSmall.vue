<template>
  <div>
    <Btn x-small outlined class="mr-4 mb-6">
      Date
      <Icon right class="text-gray-100 dark:text-gray-200">
        {{ ziCalendar }}
      </Icon>
    </Btn>
    <Btn x-small disabled outlined class="mr-4 mb-6">
      Date
      <Icon right class="text-gray-100 dark:text-gray-200">
        {{ ziCalendar }}
      </Icon>
    </Btn>
    <Btn
      x-small
      :loading="loading"
      outlined
      @click="toggleLoading"
      class="mb-6"
    >
      Date
      <Icon right class="text-gray-100 dark:text-gray-200">
        {{ ziCalendar }}
      </Icon>
    </Btn>
  </div>
  <div>
    <Btn x-small outlined class="mr-4 mb-6">
      <Icon left class="text-gray-100 dark:text-gray-200">
        {{ ziPlay }}
      </Icon>
      Start
    </Btn>
    <Btn x-small disabled outlined class="mr-4 mb-6">
      <Icon left class="text-gray-100 dark:text-gray-200">
        {{ ziPlay }}
      </Icon>
      Start
    </Btn>
    <Btn
      x-small
      :loading="loading"
      outlined
      @click="toggleLoading"
      class="mb-6"
    >
      <Icon left class="text-gray-100 dark:text-gray-200">
        {{ ziPlay }}
      </Icon>
      Start
    </Btn>
  </div>
  <div class="-mb-6">
    <Btn x-small icon outlined class="mr-4 mb-6">
      <Icon>
        {{ ziDimentions }}
      </Icon>
    </Btn>
    <Btn x-small icon disabled outlined class="mr-4 mb-6">
      <Icon>
        {{ ziDimentions }}
      </Icon>
    </Btn>
    <Btn
      x-small
      icon
      :loading="loading"
      outlined
      @click="toggleLoading"
      class="mb-6"
    >
      <Icon>
        {{ ziDimentions }}
      </Icon>
    </Btn>
  </div>
</template>

<script setup>
import { ref } from 'vue'
import { ziCalendar, ziPlay, ziDimentions } from '@zennnn/icons'

const loading = ref(false)

function toggleLoading() {
  loading.value = true
  setTimeout(() => {
    loading.value = false
  }, 2000)
}
</script>
