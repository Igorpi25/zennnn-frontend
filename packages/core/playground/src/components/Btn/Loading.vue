<template>
  <div>
    <Btn :loading="loading" @click="toggleLoading" class="mr-4 mb-6">
      <Icon left>
        {{ ziVisible }}
      </Icon>
      Text icon
    </Btn>
    <Btn :loading="loading" @click="toggleLoading" class="mr-4 mb-6">
      Text
    </Btn>
    <Btn :loading="loading" @click="toggleLoading" icon class="mb-6">
      <Icon>
        {{ ziVisible }}
      </Icon>
    </Btn>
  </div>
  <div class="-mb-6">
    <Btn :loading="loading" small @click="toggleLoading" class="mr-4 mb-6">
      <Icon left>
        {{ ziVisible }}
      </Icon>
      Text icon
    </Btn>
    <Btn :loading="loading" small @click="toggleLoading" class="mr-4 mb-6">
      Text
    </Btn>
    <Btn :loading="loading" small @click="toggleLoading" icon class="mb-6">
      <Icon>
        {{ ziVisible }}
      </Icon>
    </Btn>
  </div>
</template>

<script setup>
import { ref } from 'vue'
import { ziVisible } from '@zennnn/icons'

const loading = ref(false)

function toggleLoading () {
  loading.value = true
  setTimeout(() => {
    loading.value = false
  }, 2000)
}
</script>
