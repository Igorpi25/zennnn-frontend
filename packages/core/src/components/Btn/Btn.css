.btn {
  @apply h-12 relative rounded-lg inline-flex justify-center items-center align-middle;
  @apply transition-colors duration-100 ease-out;
  @apply select-none focus:outline-none;
  @apply px-6;
  @apply disabled:cursor-not-allowed disabled:opacity-50;
  @apply focus:ring focus:ring-blue-400;
}

.btn:focus:not(:focus-visible) {
  box-shadow: none;
  outline: 0;
}

.btn--small {
  @apply text-sm;
  @apply h-10;
  @apply px-4;
  min-width: 2.5rem;
}
.btn--x-small {
  @apply text-sm;
  @apply h-8;
  @apply px-2;
  @apply rounded;
  min-width: 2rem;
}
.btn--mini {
  @apply text-xs;
  @apply h-6;
  @apply px-1;
  @apply rounded;
  min-width: 1.5rem;
}

.btn--icon {
  @apply p-3;
}
.btn--icon.btn--small {
  @apply p-2;
}
.btn--icon.btn--x-small {
  @apply p-1;
}
.btn--icon.btn--mini {
  @apply p-0;
}

.btn--block {
  @apply w-full justify-center;
}

.btn--loading {
  @apply disabled:opacity-100;
}

.btn--primary {
  @apply text-white;
  @apply bg-blue-500;
}
.btn--primary:not(.btn--loading) {
  @apply hover:bg-blue-400 active:bg-blue-400;
  @apply disabled:bg-gray-100 disabled:text-gray-700;
}

.btn--outlined {
  @apply text-blue-500;
  @apply border border-light-gray-400 dark:border-gray-400;
}
.btn--outlined:not(.btn--loading) {
  @apply hover:text-white active:text-white hover:border-blue-400 active:border-blue-400 hover:bg-blue-400 active:bg-blue-400;
  @apply disabled:text-gray-100;
  @apply disabled:border-light-gray-400 dark:disabled:border-gray-400 disabled:bg-transparent;
}
.btn--outlined.btn--light {
  @apply dark:border-light-gray-400 dark:disabled:border-light-gray-400;
}
.btn--outlined.btn--dark {
  @apply border-gray-400 disabled:border-gray-400;
}

.btn--text {
  @apply text-blue-500;
}
.btn--text:not(.btn--loading) {
  @apply hover:text-blue-400 active:text-blue-400;
  @apply disabled:text-gray-100;
}

.btn--loading .btn__content {
  @apply opacity-0;
}

.btn__content {
  @apply flex items-center leading-none whitespace-nowrap;
}

.btn__loader {
  @apply absolute inset-0 flex items-center justify-center;
}
.btn--outlined .btn__loader {
  @apply text-blue-500;
}

.btn--dark-icon .icon {
  @apply text-gray-100 dark:text-gray-200;
}
.btn:not(:disabled).btn--dark-icon:hover .icon,
.btn:not(:disabled).btn--dark-icon:active .icon {
  @apply text-white dark:text-white;
}

.btn .icon--left {
  @apply mr-2;
}
.btn .icon--right {
  @apply ml-2;
}
.btn--x-small .icon--left,
.btn--mini .icon--left {
  @apply mr-1;
}
.btn--x-small .icon--right,
.btn--mini .icon--right {
  @apply ml-1;
}
