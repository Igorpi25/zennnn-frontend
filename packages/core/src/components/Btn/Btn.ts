import { h, ref, computed, defineComponent, VNode } from 'vue'
import { RouterLink } from 'vue-router'
import { convertToUnit } from 'vue-supp'

import Progress from '../Progress'

export default defineComponent({
  name: 'Btn',

  props: {
    tag: {
      type: String,
      default: 'button',
    },
    to: {
      type: [String, Object],
      default: undefined,
    },
    replace: Boolean,
    href: String,
    loading: Boolean,
    disabled: Boolean,
    primary: {
      type: Boolean,
      default: undefined,
    },
    block: Boolean,
    outlined: Boolean,
    xSmall: Boolean,
    mini: Boolean,
    text: Boolean,
    icon: Boolean,
    small: Boolean,
    light: Boolean,
    dark: Boolean,
    contentClass: {
      type: String,
      default: '',
    },
    minWidth: [String, Number],
    maxWidth: [String, Number],
    darkIcon: Boolean,
    retainFocusOnClick: Boolean,
  },

  setup (props, { slots }) {
    const MIN_WIDTH = 128
    const MIN_WIDTH_SMALL = 96

    const rootElement = ref<HTMLElement>()

    const isDisabled = computed(() => {
      return props.disabled || props.loading
    })

    const classes = computed(() => {
      return {
        btn: true,
        'btn--primary': !props.text && !props.outlined && props.primary !== false,
        'btn--outlined': !props.text && props.outlined,
        'btn--block': props.block,
        'btn--text': props.text,
        'btn--icon': props.icon,
        'btn--dark-icon': props.darkIcon,
        'btn--small': props.small,
        'btn--x-small': props.xSmall,
        'btn--mini': props.mini,
        'btn--loading': props.loading,
        'btn--disabled': props.disabled,
        'btn--light': props.light,
        'btn--dark': props.dark,
      }
    })

    const styles = computed(() => {
      let minWidth = null
      if (!props.icon && !props.text && !props.xSmall && !props.mini) {
        minWidth = props.small ? MIN_WIDTH_SMALL : MIN_WIDTH
      }
      return {
        minWidth: convertToUnit(props.minWidth || minWidth),
        maxWidth: convertToUnit(props.maxWidth),
      }
    })

    const genContent = () => {
      const data = {
        class: {
          btn__content: true,
          [props.contentClass.trim()]: true,
        },
      }
      return h('span', data, slots.default?.())
    }

    const genLoader = () => {
      return h('span', {
        class: 'btn__loader',
      }, slots.loader
        ? slots.loader()
        : h(Progress, {
          indeterminate: true,
          size: 24,
          width: 2,
        }))
    }

    const genRouterLink = (data: any, children: VNode[]) => {
      Object.assign(data, {
        to: props.to,
        replace: props.replace,
      })
      return h(RouterLink, data, {
        default: () => children,
      })
    }

    const genButton = (data: any, children: VNode[]) => {
      const tag = (props.href && 'a') || props.tag || 'div'
      if (tag === 'a') {
        // TODO: data href="null" not remove href from element
        // not set href on disabled or loading
        if (props.href && !isDisabled.value) {
          data.href = props.href
        }
      }
      if (tag === 'button') {
        data.disabled = isDisabled.value || null
      }
      data.onClick = () => {
        !props.retainFocusOnClick && rootElement.value?.blur()
      }
      return h(tag, data, children)
    }

    return {
      rootElement,
      classes,
      styles,
      genContent,
      genLoader,
      genRouterLink,
      genButton,
    }
  },

  render () {
    const children = [
      this.genContent(),
      this.loading && this.genLoader(),
    ] as VNode[]
    const data = {
      ref: 'rootElement',
      class: this.classes,
      style: this.styles,
    }
    if (this.to) {
      return this.genRouterLink(data, children)
    } else {
      return this.genButton(data, children)
    }
  },
})
