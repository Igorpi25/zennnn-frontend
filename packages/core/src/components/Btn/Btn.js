import { h, ref, computed } from 'vue'
import { RouterLink } from 'vue-router'
import { convertToUnit } from 'vue-supp'

import Progress from '../Progress'

import './Btn.css'

export default {
  name: 'Btn',

  props: {
    tag: {
      type: String,
      default: 'button',
    },
    to: {
      type: [String, Object],
      default: undefined,
    },
    append: Boolean,
    replace: Boolean,
    href: String,
    loading: Boolean,
    disabled: Boolean,
    primary: {
      type: Boolean,
      default: undefined,
    },
    block: Boolean,
    outlined: Boolean,
    xSmall: Boolean,
    mini: Boolean,
    text: Boolean,
    icon: Boolean,
    small: Boolean,
    light: Boolean,
    dark: Boolean,
    contentClass: {
      type: String,
      default: '',
    },
    minWidth: [String, Number],
    maxWidth: [String, Number],
    darkIcon: Boolean,
  },

  setup (props, { slots }) {
    const MIN_WIDTH = 128
    const MIN_WIDTH_SMALL = 96

    const rootElement = ref(null)

    const isDisabled = computed(() => {
      return props.disabled || props.loading
    })

    const classes = computed(() => {
      return {
        btn: true,
        'btn--primary': !props.text && !props.outlined && props.primary !== false,
        'btn--outlined': !props.text && props.outlined,
        'btn--block': props.block,
        'btn--text': props.text,
        'btn--icon': props.icon,
        'btn--dark-icon': props.darkIcon,
        'btn--small': props.small,
        'btn--x-small': props.xSmall,
        'btn--mini': props.mini,
        'btn--loading': props.loading,
        'btn--disabled': props.disabled,
        'btn--light': props.light,
        'btn--dark': props.dark,
      }
    })

    const styles = computed(() => {
      let minWidth = null
      if (!props.icon && !props.text && !props.xSmall && !props.mini) {
        minWidth = props.small ? MIN_WIDTH_SMALL : MIN_WIDTH
      }
      return {
        minWidth: convertToUnit(props.minWidth || minWidth),
        maxWidth: convertToUnit(props.maxWidth),
      }
    })

    const genContent = () => {
      const data = {
        class: {
          btn__content: true,
          [props.contentClass.trim()]: true,
        },
      }
      return h('span', data, slots.default ? slots.default() : null)
    }

    const genLoader = () => {
      return h('span', {
        class: 'btn__loader',
      }, slots.loader
        ? slots.loader()
        : h(Progress, {
          indeterminate: true,
          size: 24,
          width: 2,
        }))
    }

    const genRouterLink = (data, children) => {
      Object.assign(data, {
        to: props.to,
        append: props.append,
        replace: props.replace,
      })
      return h(RouterLink, data, {
        default: () => {
          return [children]
        },
      })
    }

    const genButton = (data, children) => {
      const tag = (props.href && 'a') || props.tag || 'div'
      if (tag === 'a') {
        // TODO: data href="null" not remove href from element
        // not set href on disabled or loading
        if (props.href && !isDisabled.value) {
          data.href = props.href
        }
      }
      if (tag === 'button') {
        data.disabled = isDisabled.value || null
      }
      data.onClick = () => {
        rootElement.value.blur()
      }
      return h(tag, data, children)
    }

    return {
      rootElement,
      classes,
      styles,
      genContent,
      genLoader,
      genRouterLink,
      genButton,
    }
  },

  render () {
    const children = [
      this.genContent(),
      this.loading && this.genLoader(),
    ]
    const data = {
      ref: 'rootElement',
      class: this.classes,
      style: this.styles,
    }
    if (this.to) {
      return this.genRouterLink(data, children)
    } else {
      return this.genButton(data, children)
    }
  },
}
