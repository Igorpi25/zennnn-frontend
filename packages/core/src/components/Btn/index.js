import Btn from './Btn'

export { Btn }
export default Btn
