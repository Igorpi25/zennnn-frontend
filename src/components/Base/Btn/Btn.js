import { h, ref, computed } from 'vue'
import { RouterLink } from 'vue-router'

import './Btn.css'

import Progress from '../Progress'

import { convertToUnit } from '../../../utils/convertToUnit'

export default {
  name: 'Btn',

  props: {
    tag: {
      type: String,
      default: 'button',
    },
    to: {
      type: [String, Object],
      default: undefined,
    },
    append: Boolean,
    replace: Boolean,
    href: String,
    loading: Boolean,
    disabled: Boolean,
    primary: Boolean,
    block: Boolean,
    outlined: Boolean,
    borderless: Boolean,
    text: Boolean,
    icon: Boolean,
    small: Boolean,
    light: Boolean,
    dark: Boolean,
    contentClass: {
      type: String,
      default: '',
    },
    minWidth: [String, Number],
    maxWidth: [String, Number],
  },

  setup (props, { slots }) {
    const MIN_WIDTH = 120
    const MIN_WIDTH_SMALL = 100

    const rootElement = ref(null)

    const isDisabled = computed(() => {
      return props.disabled || props.loading
    })

    const classes = computed(() => {
      const outlined = props.outlined || props.borderless
      return {
        btn: true,
        'btn--primary': props.primary || !outlined,
        'btn--outlined': outlined,
        'btn--borderless': props.borderless,
        'btn--block': props.block,
        'btn--text': props.text,
        'btn--icon': props.icon,
        'btn--small': props.small,
        'btn--loading': props.loading,
        'btn--disabled': props.disabled,
        'btn--light': props.light,
        'btn--dark': props.dark,
      }
    })

    const styles = computed(() => {
      let minWidth = null
      if (!props.icon && !props.text) {
        minWidth = props.small ? MIN_WIDTH_SMALL : MIN_WIDTH
      }
      return {
        minWidth: convertToUnit(props.minWidth || minWidth),
        maxWidth: convertToUnit(props.maxWidth),
      }
    })

    const genContent = () => {
      const data = {
        class: {
          btn__content: true,
          [props.contentClass.trim()]: true,
        },
      }
      return h('span', data, slots.default ? slots.default() : null)
    }

    const genLoader = () => {
      return h('span', {
        class: 'btn__loader',
      }, slots.loader
        ? slots.loader()
        : h(Progress, {
          indeterminate: true,
          size: 24,
          width: 2,
        }))
    }

    const genRouterLink = (data, children) => {
      Object.assign(data, {
        to: props.to,
        append: props.append,
        replace: props.replace,
      })
      return h(RouterLink, data, {
        default: () => {
          return [children]
        },
      })
    }

    const genButton = (data, children) => {
      const tag = (props.href && 'a') || props.tag || 'div'
      if (tag === 'a') {
        // TODO: data href="null" not remove href from element
        // not set href on disabled or loading
        if (props.href && !isDisabled.value) {
          data.href = props.href
        }
      }
      if (tag === 'button') {
        data.disabled = isDisabled.value || null
      }
      data.onClick = () => {
        rootElement.value.blur()
      }
      return h(tag, data, children)
    }

    return {
      rootElement,
      classes,
      styles,
      genContent,
      genLoader,
      genRouterLink,
      genButton,
    }
  },

  render () {
    const children = [
      this.genContent(),
      this.loading && this.genLoader(),
    ]
    const data = {
      ref: 'rootElement',
      class: this.classes,
      style: this.styles,
    }
    if (this.to) {
      return this.genRouterLink(data, children)
    } else {
      return this.genButton(data, children)
    }
  },
}
