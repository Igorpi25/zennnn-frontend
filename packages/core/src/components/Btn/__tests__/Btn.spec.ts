import { mount, VueWrapper } from '@vue/test-utils'
import { createRouter, createWebHistory } from 'vue-router'
import Btn from '../Btn'

describe('Btn.ts', () => {
  let mountFunction: (options?: Record<string, unknown>) => VueWrapper<any>

  beforeEach(() => {
    mountFunction = (options = {}) => {
      return mount(Btn, {
        ...options,
      })
    }
  })

  // Render

  it('should render component with default slot and match snapshot', () => {
    const wrapper = mountFunction({
      slots: {
        default: () => 'Button',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with loader slot and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        loading: true,
      },
      slots: {
        loader: () => '<span>Loading</span>',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with loader and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        loading: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render an <a> tag when using href prop and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        href: 'http://example.com',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render specified tag when using tag prop and match snapshot', () => {
    const wrapper = mountFunction({
      props: {
        tag: 'a',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render router-link when using to prop and match snapshot', () => {
    const router = createRouter({
      history: createWebHistory(),
      routes: [
        {
          path: '/',
          component: {
            template: 'Welcome to home',
          },
        },
        {
          path: '/foo',
          component: {
            template: 'Welcome to foo',
          },
        },
      ],
    })

    const wrapper = mountFunction({
      global: {
        plugins: [router],
      },
      props: {
        to: '/foo',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  // Html attributes

  it('should inherit attributes like target, tabindex etc.', () => {
    const wrapper = mountFunction({
      attrs: {
        target: '_self',
      },
      props: {
        tag: 'a',
      },
    })

    expect(wrapper.attributes('target')).toBe('_self')
  })

  it('should not have href attribute on disabled <a> tag', () => {
    const wrapper = mountFunction({
      props: {
        tag: 'a',
        disabled: true,
      },
    })

    expect(wrapper.attributes('href')).toBe(undefined)
  })

  it('should have disabled attribute on disabled and loading prop', async () => {
    const wrapper = mountFunction({
      props: {
        disabled: true,
      },
    })

    expect(wrapper.attributes('disabled')).toBe('')

    await wrapper.setProps({ disabled: true, loading: true })

    expect(wrapper.attributes('disabled')).toBe('')
  })

  it('should have min-width and max-width styles on minWidth and maxWidth prop', () => {
    const wrapper = mountFunction({
      props: {
        minWidth: 200,
        maxWidth: 250,
      },
    })

    expect((wrapper.element as HTMLElement).style.minWidth).toBe('200px')
    expect((wrapper.element as HTMLElement).style.maxWidth).toBe('250px')
  })

  it('should have correct default min-width styles', async () => {
    const wrapper = mountFunction({
      props: {},
    })

    expect((wrapper.element as HTMLElement).style.minWidth).toBe('128px')

    await wrapper.setProps({ small: true })

    expect((wrapper.element as HTMLElement).style.minWidth).toBe('96px')
  })

  it('should not have default min-width style on icon/text/xSmall/mini props', async () => {
    const wrapper = mountFunction({
      props: {
        icon: true,
      },
    })

    expect((wrapper.element as HTMLElement).style.minWidth).toBe('')

    await wrapper.setProps({ text: true })

    expect((wrapper.element as HTMLElement).style.minWidth).toBe('')

    await wrapper.setProps({ xSmall: true })

    expect((wrapper.element as HTMLElement).style.minWidth).toBe('')

    await wrapper.setProps({ mini: true })

    expect((wrapper.element as HTMLElement).style.minWidth).toBe('')
  })

  // Class modifiers

  it('should have btn--block class when using block prop', () => {
    const wrapper = mountFunction({
      props: {
        block: true,
      },
    })

    expect(wrapper.classes('btn--block')).toBe(true)
  })

  it('should have btn--small class when using small prop', () => {
    const wrapper = mountFunction({
      props: {
        small: true,
      },
    })

    expect(wrapper.classes('btn--small')).toBe(true)
  })

  it('should have btn--x-small class when using x-small prop', () => {
    const wrapper = mountFunction({
      props: {
        xSmall: true,
      },
    })

    expect(wrapper.classes('btn--x-small')).toBe(true)
  })

  it('should have btn--mini class when using mini prop', () => {
    const wrapper = mountFunction({
      props: {
        mini: true,
      },
    })

    expect(wrapper.classes('btn--mini')).toBe(true)
  })

  it('should have btn--primary class by default or when using primary prop', async () => {
    const wrapper = mountFunction({
      props: {},
    })

    expect(wrapper.classes('btn--primary')).toBe(true)

    await wrapper.setProps({ primary: true })

    expect(wrapper.classes('btn--primary')).toBe(true)
  })

  it('should not have btn--primary class when using text, outlined prop or primary=false', async () => {
    const wrapper = mountFunction({
      props: {
        primary: true,
        text: true,
      },
    })

    expect(wrapper.classes('btn--primary')).toBe(false)

    await wrapper.setProps({ primary: true, outlined: true })

    expect(wrapper.classes('btn--primary')).toBe(false)

    await wrapper.setProps({ primary: false })

    expect(wrapper.classes('btn--primary')).toBe(false)
  })

  it('should have btn--outlined class when using outlined prop', () => {
    const wrapper = mountFunction({
      props: {
        outlined: true,
      },
    })

    expect(wrapper.classes('btn--outlined')).toBe(true)
  })

  it('should not have btn--outlined class when using text or link prop', () => {
    const wrapper = mountFunction({
      props: {
        text: true,
      },
    })

    expect(wrapper.classes('btn--outlined')).toBe(false)

    wrapper.setProps({ link: true })

    expect(wrapper.classes('btn--outlined')).toBe(false)
  })

  it('should have correct custom classes', () => {
    const wrapper = mountFunction({
      propsData: {
        class: 'foo',
      },
    })

    expect(wrapper.element.className).toBe('btn btn--primary foo')
  })
})
