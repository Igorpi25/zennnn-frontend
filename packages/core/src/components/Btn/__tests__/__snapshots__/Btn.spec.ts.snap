// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn.ts should render an <a> tag when using href prop and match snapshot 1`] = `
<a class="btn btn--primary" style="min-width: 128px;" href="http://example.com"><span class="btn__content"></span>
  <!---->
</a>
`;

exports[`Btn.ts should render component with default slot and match snapshot 1`] = `
<button class="btn btn--primary" style="min-width: 128px;"><span class="btn__content">Button</span>
  <!---->
</button>
`;

exports[`Btn.ts should render component with loader and match snapshot 1`] = `<button class="btn btn--primary btn--loading" style="min-width: 128px;" disabled=""><span class="btn__content"></span><span class="btn__loader"><div role="progressbar" aria-valuemin="0" aria-valuemax="100" class="progress progress--indeterminate" style="height: 24px; width: 24px;"><svg style="transform: rotate(0deg);" xmlns="http://www.w3.org/2000/svg" viewBox="21.81818181818182 21.81818181818182 43.63636363636364 43.63636363636364"><!----><circle class="progress__overlay" fill="transparent" cx="43.63636363636364" cy="43.63636363636364" r="20" stroke-width="3.6363636363636367" stroke-dasharray="125.664" stroke-dashoffset="125.66370614359172px"></circle></svg><div class="progress__info"></div></div></span></button>`;

exports[`Btn.ts should render component with loader slot and match snapshot 1`] = `<button class="btn btn--primary btn--loading" style="min-width: 128px;" disabled=""><span class="btn__content"></span><span class="btn__loader">&lt;span&gt;Loading&lt;/span&gt;</span></button>`;

exports[`Btn.ts should render router-link when using to prop and match snapshot 1`] = `
<a href="/foo" class="btn btn--primary" style="min-width: 128px;"><span class="btn__content"></span>
  <!---->
</a>
`;

exports[`Btn.ts should render specified tag when using tag prop and match snapshot 1`] = `
<a class="btn btn--primary" style="min-width: 128px;"><span class="btn__content"></span>
  <!---->
</a>
`;
